scripts/MAZE-EN/c100206006.lua

```diff
@@ -0,0 +1,75 @@
+--Gate Guardian of Water and Thunder
+--scripted by JoyJ
+function c100206006.initial_effect(c)
+	--fusion material
+	c:EnableReviveLimit()
+	aux.AddFusionProcCode2(c,25955164,98434877,true,true)
+	aux.AddContactFusionProcedure(c,Card.IsAbleToRemoveAsCost,LOCATION_ONFIELD+LOCATION_HAND+LOCATION_REMOVED,0,Duel.Remove,POS_FACEUP,REASON_COST)
+	--atk 0
+	local e1=Effect.CreateEffect(c)
+	e1:SetDescription(aux.Stringid(100206006,0))
+	e1:SetCategory(CATEGORY_ATKCHANGE)
+	e1:SetProperty(EFFECT_FLAG_CARD_TARGET+EFFECT_FLAG_DAMAGE_STEP)
+	e1:SetType(EFFECT_TYPE_QUICK_O)
+	e1:SetCode(EVENT_FREE_CHAIN)
+	e1:SetRange(LOCATION_MZONE)
+	e1:SetCountLimit(2,100206006)
+	e1:SetTarget(c100206006.atktg)
+	e1:SetOperation(c100206006.atkop)
+	c:RegisterEffect(e1)
+	--special summon
+	local e2=Effect.CreateEffect(c)
+	e2:SetDescription(aux.Stringid(100206006,2))
+	e2:SetCategory(CATEGORY_SPECIAL_SUMMON)
+	e2:SetType(EFFECT_TYPE_SINGLE+EFFECT_TYPE_TRIGGER_O)
+	e2:SetCode(EVENT_LEAVE_FIELD)
+	e2:SetProperty(EFFECT_FLAG_DELAY+EFFECT_FLAG_DAMAGE_STEP)
+	e2:SetCondition(c100206006.spcon)
+	e2:SetTarget(c100206006.sptg)
+	e2:SetOperation(c100206006.spop)
+	c:RegisterEffect(e2)
+end
+function c100206006.atkfilter(c)
+	return c:IsFaceup() and c:GetAttack()>0
+end
+function c100206006.atktg(e,tp,eg,ep,ev,re,r,rp,chk,chkc)
+	if chkc then return chkc:IsControler(1-tp) and chkc:IsLocation(LOCATION_MZONE) and c100206006.atkfilter(chkc) end
+	if chk==0 then return Duel.IsExistingTarget(c100206006.atkfilter,tp,0,LOCATION_MZONE,1,nil)
+		and c:GetFlagEffect(100206006)==0 end
```

Comment:
redtext: `attempt to index a nil value (global 'c')`
```suggestion
		and e:GetHandler():GetFlagEffect(100206006)==0 end
```